File: backend/app/schemas/bulk_analysis.py
from pydantic import BaseModel, Field

from app.schemas.analysis import AnalysisResponse


class InvalidIPEntry(BaseModel):
    value: str
    reason: str


class BulkAnalysisResponse(BaseModel):
    filename: str
    total_entries: int
    valid_count: int
    invalid_count: int
    duplicate_count: int

    results: list[AnalysisResponse] = Field(
        default_factory=list
    )

    invalid_entries: list[InvalidIPEntry] = Field(
        default_factory=list
    )

File: frontend/src/types/bulkAnalysis.ts
import type {
  AnalysisResponse,
} from "./analysis";


export interface InvalidIPEntry {
  value: string;
  reason: string;
}


export interface BulkAnalysisResponse {
  filename: string;
  total_entries: number;
  valid_count: number;
  invalid_count: number;
  duplicate_count: number;
  results: AnalysisResponse[];
  invalid_entries: InvalidIPEntry[];
}